runtime/src/chpl-mem.c

```diff
@@ -93,15 +93,15 @@ int chpl_posix_memalign(void** ptr, size_t alignment, size_t size) {
 
 void* chpl_valloc(size_t size)
 {
-  return chpl_memalign(chpl_getHeapPageSize(), size);
+  return chpl_memalign(chpl_getSysPageSize(), size);
```

Comment:
Yeah, it's a little tricky because if you're allocating (presumably from a heap) then you'd think you want the heap page size, but if you're going to do something system-like with the space (like an mmap(2)) then you want the system page size. It would probably be best to do this separately.